### DISCUSSION TITLE
Allow getting type information from file without requiring it

### DISCUSSION THREAD

**Discussion by Hedwig7s** [teal-language/tl#790] [Q&A] [answered]
Sometimes you have a module that uses types from another module but doesn't actually need the functionality for it (for example: a module that defines functions that use types from another module which is called elsewhere)

Right now there are 2 options that I know of (if there's another way I'd love to know):
Make a type declaration file (which isn't preferable if you want to define the types where they're made and can be inconvenient)
Forward-declare type without any information (provides no type information within file)

If a better way to handle this situation could be considered that would be great

**Comment by hishamhm** [comment, marked as answer]
You can do that today with

```
local type mod = require("mod")
```

This will load the type info without outputting a `require()` call in the generated code.

**Reply by Hedwig7s** [reply]
That should probably be in tutorial.md (either that or I'm blind)

**Reply by hishamhm** [reply]
@Hedwig7s I think it isn't! a docs PR with a contribution is most welcome!!
